NebulaClient/PacketProcessors/Factory/Foundation/FoundationBuildUpdateProcessor.cs

```diff
@@ -0,0 +1,60 @@
+﻿using NebulaModel.Attributes;
+using NebulaModel.DataStructures;
+using NebulaModel.Networking;
+using NebulaModel.Packets.Factory;
+using NebulaModel.Packets.Processors;
+using NebulaWorld.Factory;
+using UnityEngine;
+
+namespace NebulaClient.PacketProcessors.Factory.Foundation
+{
+    [RegisterPacketProcessor]
+    class FoundationBuildUpdateProcessor : IPacketProcessor<FoundationBuildUpdatePacket>
+    {
+        public void ProcessPacket(FoundationBuildUpdatePacket packet, NebulaConnection conn)
+        {
+            //Check if client has loaded planet
+            PlanetData planet = GameMain.galaxy.PlanetById(packet.PlanetId);
+            PlanetFactory factory = planet?.factory;
+            if (factory != null)
+            {
+                Vector3[] reformPoints = new Vector3[100];
```

Comment:
Looks like something that could be cached for the processor, since there's no concurrent users.

Reply:
This is the "output" variable that will have the result from this method
```cs
ComputeFlattenTerrainReform(reformPoints, ...
```

But I guess I can cache the allocated array so it will not be allocated each time the packet is received and just clean the array with empty values.

Reply:
`ArrayPool<T>` is only available in newer target frameworks, sadly. This is not a big issue btw, it's just that it might cause gen0 GCs in done frequently.